base/pmap.jl

```diff
@@ -190,13 +188,11 @@ function process_batch_errors!(p, f, results, on_error, retry_on, retry_n, retry
         if length(reprocess) > 0
             errors = [x[2] for x in reprocess]
             exceptions = [x.ex for x in errors]
-            if (retry_n > 0) && all([retry_on(ex) for ex in exceptions])
-                retry_n = retry_n - 1
+            state = start(retry_delays)
+            if (length(retry_delays) > 0) &&
+                    (retry_check==nothing || all([retry_check(state,ex)[2] for ex in exceptions]))
                 error_processed = pmap(p, f, [x.data for x in errors];
-                                                    on_error=on_error,
-                                                    retry_on=retry_on,
-                                                    retry_n=retry_n,
-                                                    retry_max_delay=retry_max_delay)
+                        on_error = on_error, retry_delays = retry_delays[2:end], retry_check = retry_check)
```

Comment:
I think this may not work if retry_delays is of type ExponentialBackOff? I think the retry logic in pmap will need a different approach. We can tackle it as a separate issue.